src/ensembl/src/content/app/entity-viewer/shared/helpers/tests/transcripts-filter.test.ts

```diff
@@ -0,0 +1,136 @@
+import { filterTranscriptsBySOTerm } from '../transcripts-filter';
+
+import { createTranscript } from 'tests/fixtures/entity-viewer/transcript';
+
+/* Creating filters with different filter set to true */
+const proteinCodingfilters = {
+  protein_coding: true,
+  retained_intron: false,
+  processed_transcript: false,
+  nonsense_mediated_decay: false
+};
+
+const retainedIntronfilters = {
+  protein_coding: false,
+  retained_intron: true,
+  processed_transcript: false,
+  nonsense_mediated_decay: false
+};
+
+const processedTranscriptfilters = {
+  protein_coding: false,
+  retained_intron: false,
+  processed_transcript: true,
+  nonsense_mediated_decay: false
+};
+
+const nonsenseMediatedDecayfilters = {
+  protein_coding: false,
+  retained_intron: false,
+  processed_transcript: false,
+  nonsense_mediated_decay: true
+};
+
+/* Creating dummy transcripts with different different so_term  to test so_term filtering */
+const createProteinCodingTranscript = () => {
+  const transcript = createTranscript();
+  transcript.so_term = 'protein_coding';
+  return transcript;
+};
+
+const createProcessedTranscript = () => {
+  const transcript = createTranscript();
+  transcript.so_term = 'processed_transcript';
+  return transcript;
+};
+
+const createRetainedIntronTranscript = () => {
+  const transcript = createTranscript();
+  transcript.so_term = 'retained_intron';
+  return transcript;
+};
+
+const createNonsenseMediatedDecayTranscript = () => {
+  const transcript = createTranscript();
+  transcript.so_term = 'nonsense_mediated_decay';
+  return transcript;
+};
+
+const ProteinCodingTranscript = createProteinCodingTranscript();
+const ProcessedTranscript = createProcessedTranscript();
+const RetainedIntronTranscript = createRetainedIntronTranscript();
+const NonsenseMediatedDecayTranscript = createNonsenseMediatedDecayTranscript();
+
+describe('filter transcripts by so_term', () => {
```

Comment:
**Minor point:** the string passed to the `describe` function is usually the name of the function or a class that is being tested. In this case `describe('filterTranscriptsBySOTerm', () => {...})`. It even reads like an English sentence: `describe('a thing') ... it<i.e. that thing you mentioned in the describe block>('does so-and-so')`.

**Major point:** The tests in this file are very repetitive. We don't really need to test 4 different specific so_term strings.

If we want to test filtering transcripts by so_term, then we only need a single test. In pseudocode, the test would look something like this:

- Generate a bunch of transcripts (e.g. five transcripts)
- Choose a couple of transcripts randomly out of that bunch and add some unique string to their so_term field. It doesn't have to be a real so_term, because our filter function doesn't care about whether so_terms are real.
- Apply your filter function to this bunch: `filterTranscriptsBySOTerm(transcripts, { my_unique_string: true })`
- Check that the output from the filter function contains only transcripts that have the so_term field set to that unique string.